### DISCUSSION TITLE
How much of the track data is committed to the SQLite3 database?

### DISCUSSION THREAD

**Discussion by werame** [audacity/audacity#2438] [Development]
I see Audacity has a transactional system based on SQLite3 for undos, etc. Can someone give me an overview, either here or point to some pre-existing document, of how much that convers? Are track data blobs committed to the DB, or are they external, i..e. the transactions don't cover actual data, only project "metadata" in some broad sense.

**Comment by crsib** [comment]
All data is stored in the SQlite

**Reply by werame** [reply]
So manual rollbacks [like this](https://github.com/audacity/audacity/blob/b58d7711b405aff9009bfb487159e76c8d99beed/src/effects/Effect.cpp#L1036) removal of a track insertion (in a case of plug-in run failure) are unnecessary now? By "unnecessary" I mean: can it be replaced with some call that does generic transaction rollback, and which would also be able to handle, e.g. multi-track insertions (which that piece I linked to cannot handle, as-is)?

I see from the [git blame](https://github.com/audacity/audacity/blame/b58d7711b405aff9009bfb487159e76c8d99beed/src/effects/Effect.cpp#L1045) that the `trans.Commit` line is newer (as I expected) and was almost certainly added with the SQLite3 batch of commits, but the "manual rollback" code above it (`mTracks->Remove(newTrack);`) is 4 years old.
